galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt

```diff
@@ -2028,30 +2029,28 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
   }
 
   protected fun fetchLookup(tableIndex: Int, currentField: VField) {
-    val table = object : Table(tables!![tableIndex].tableName) {}
+    val table = tables!![tableIndex]
     val columns = mutableListOf<Column<*>>()
     val conditions = mutableListOf<Op<Boolean>>()
 
-    for (i in fields.indices) {
-      val f = fields[i]
-
-      if (f !== currentField && f.lookupColumn(tableIndex) != null && !f.isLookupKey(tableIndex)) {
-        f.setNull(activeRecord)
+    fields.forEach { field ->
+      if (field !== currentField && field.lookupColumn(tableIndex) != null && !field.isLookupKey(tableIndex)) {
+        field.setNull(activeRecord)
       }
     }
 
-    for (i in fields.indices) {
-      val column = fields[i].lookupColumn(tableIndex)
+    fields.forEach {field ->
+      val column = field.lookupColumn(tableIndex)
 
       if (column != null) {
         columns.add(column)
       }
 
-      if (fields[i] == currentField || fields[i].isLookupKey(tableIndex)) {
-        val condition = fields[i].getSearchCondition()
+      if (field == currentField || field.isLookupKey(tableIndex)) {
+        val condition = field.getSearchCondition(column)
 
         // TODO: 10/12/2020 FIX this !
-        if (condition == null || fields[i].lookupColumn(tableIndex)!!.condition() !is EqOp) {
+        if (condition == null || condition !is EqOp) {
```

Comment:
can you check `condition !is EqOp` 
because condition is an **Op<Boolean** it's not an operator